hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/cache/topic/NotificationBasedTopicsCache.java

```diff
@@ -7,22 +7,25 @@
 import pl.allegro.tech.hermes.domain.notifications.InternalNotificationsBus;
 import pl.allegro.tech.hermes.domain.notifications.TopicCallback;
 import pl.allegro.tech.hermes.domain.topic.TopicRepository;
+import pl.allegro.tech.hermes.frontend.blacklist.BlacklistZookeeperNotifyingCache;
+import pl.allegro.tech.hermes.frontend.blacklist.TopicBlacklistCallback;
 import pl.allegro.tech.hermes.frontend.metric.CachedTopic;
 
 import java.util.Optional;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.*;
 
-public class NotificationBasedTopicsCache implements TopicCallback, TopicsCache {
+public class NotificationBasedTopicsCache implements TopicCallback, TopicsCache, TopicBlacklistCallback {
 
     private final ConcurrentMap<String, CachedTopic> topicCache = new ConcurrentHashMap<>();
+    private final ConcurrentSkipListSet<String> blacklistedTopics = new ConcurrentSkipListSet<>();
```

Comment:
why not `Collections.newSetFromMap(new ConcurrentHashMap())`? this looks like an overkill, i just want set